lib/extensions/src/lib/services/extension-loader.service.ts

```diff
@@ -42,6 +42,10 @@ export class ExtensionLoaderService {
                         config = JSON.parse(override);
                     }
 
+                    if (extensions && extensions.length) {
+                        config.$references.push(...this.filterIgnoredExtensions(extensions, config));
```

Comment:
If we push, there is no way to override the the order. As I see, here we need to do is something similar to this:\\\\\\\\\

```js
// No $references defined in the app.extensions.json, so we can continue the dynamically registered ones
// coming from the extensions variable, paying attention to the ignored ones as well
if (!config.$references || !config.$references.length) {
	config.$references = this.filterIgnoredExtensions(extensions, config);
}
// User defined their own array of $references, we should proceed with that, paying attention
// to the ignored ones as well
else {
	config.$references = this.filterIgnoredExtensions(config.$references, config);
}
```

For this, the application (ACA/ADW) should not have anything in their $references by default, which means, we need to modify even ACA to dynamically register its plugins either.

Reply:
the  `push` is just to load the registered plugins.  Isn't the order set by the subsequent logic https://github.com/Alfresco/alfresco-ng2-components/pull/6348/files/a0fb40821839ddf565aa5f4b33e3cc885daac35f#diff-c6a1c872a637e729391870f0795194fb846e560b34090f242e98be038057977bL50-L54 ?